api/cluster.go

```diff
@@ -358,54 +358,56 @@ func GetNodePools(c *gin.Context) {
 		return
 	}
 
+	nodePoolStatus := make(map[string]*pkgCluster.ActualNodePoolStatus)
+	clusterDesiredResources := make(map[string]float64)
 	clusterTotalResources := make(map[string]float64)
-	nodePoolCounts, err := getActualNodeCounts(commonCluster)
-	if err != nil {
-		log.Error(emperror.Wrap(err, "Error getting actual node count for node pool info"))
-		c.JSON(http.StatusServiceUnavailable, pkgCommon.ErrorResponse{
-			Code:    http.StatusBadRequest,
-			Message: "Error getting actual node count for node pool info",
-			Error:   err.Error(),
-		})
-		return
-	}
 
-	nodePoolStatus := make(map[string]*pkgCluster.ActualNodePoolStatus)
+	autoScaleEnabled := commonCluster.GetScaleOptions() != nil && commonCluster.GetScaleOptions().Enabled
+	if autoScaleEnabled {
 
-	headNodePoolName := viper.GetString(config.PipelineHeadNodePoolName)
-	for nodePoolName, nodePool := range clusterStatus.NodePools {
-		if nodePoolName == headNodePoolName {
-			continue
+		nodePoolCounts, err := getActualNodeCounts(commonCluster)
+		if err != nil {
+			log.Error(emperror.Wrap(err, "Error getting actual node count for node pool info"))
+			c.JSON(http.StatusServiceUnavailable, pkgCommon.ErrorResponse{
```

Comment:
Isn't there an error handler already registered upstream in the callstack that takes care of sending an ErrorResponse back to the client?

Reply:
Fixed, thanks @stoader !